mitodl/open-discussions
# add channel about page frontend

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #1778 
closes #1623

#### What's this PR do?

This provides the frontend portion of the channel about page. That includes a new page, facilities to edit an save the about field on the channel, and a refactor of the channel routing logic to facilitate both adding a new page and removing some redundancy.


#### How should this be manually tested?

Make sure that all routes under `/c/channelName` work correctly and there are no regressions anywhere (I did some pretty extensive testing but there is a chance I missed something).

Ensure that, if you're a moderator user, you see an 'about' thing in the channel-level navbar.

if you go there you should be able to edit your channel's about document.

If you're not a moderator user you shouldn't see the link for the moderator page on channels which do not have the document saved, and you should see it on those that do. For non-moderator users the channel document should display nicely.

#### Screenshots (if appropriate)

'about' button shows up in nav always for moderators:

![aboutbutton](https://user-images.githubusercontent.com/6207644/52592191-bce6a800-2e13-11e9-8527-6a93e76be6ac.png)
![aboutbuttonmod2](https://user-images.githubusercontent.com/6207644/52592209-c8d26a00-2e13-11e9-9e23-cd82a7f5a7af.png)

(first channel has an about document saved, second does not, but the about link still shows up on the second one b/c my user is a mod)

for anonymous or non-moderator users it will only show up if the channel has a saved about document:

![aboutbuttonmod23](https://user-images.githubusercontent.com/6207644/52592270-f4555480-2e13-11e9-943c-c2d3adfec45c.png)

vs

![aboutbuttonmod23fffdfdf](https://user-images.githubusercontent.com/6207644/52592274-f8817200-2e13-11e9-936a-ef56bf94956e.png)

the page itself looks like this for mods:

![aboutbuttonmod](https://user-images.githubusercontent.com/6207644/52592287-02a37080-2e14-11e9-9f72-94c2e7391b72.png)

with some minimal UI for editing and saving, and like this for non-mods:

![aboutbuttonmod23ff](https://user-images.githubusercontent.com/6207644/52592296-0cc56f00-2e14-11e9-8a50-1db745b20206.png)

mob:

![aboutbuttonmod23fffdfdfmobfff](https://user-images.githubusercontent.com/6207644/52592458-6af25200-2e14-11e9-8759-98f3c3f5da03.png)
![aboutbuttonmod23fffdfdfmob](https://user-images.githubusercontent.com/6207644/52592459-6af25200-2e14-11e9-9f40-bd646044c4ed.png)
